modAL/datamanager.py

```diff
@@ -0,0 +1,136 @@
+import typing
+from typing import List, Tuple, Any, Union, Optional,Generic,TypeVar
+import numpy as np
+Label = Tuple[int, np.dtype]
+LabelList = List[Label]
+Sources = List[Any]
+
+
+class DataManager:
+    def __init__(
+        self,
+        features: np.ndarray,
+        labels_dtype: Optional[np.dtype] = None,
+        sources: Optional[Sources] = None,
+    ):
+        """
+
+        When doing active learning we have our Original Data (OD) Labeled Data [LD] and Unlabeled Data [UD]
+        where UD and LD are subsets of OD.
+        The active learner operates on UD and returns indexes relative to it. We want to store those indices with respect
+        to OD, and sometimes see the subset of labels of LD. (The subset of labels of UD is Null)
+
+        That's a fancy way of saying there is a lot book keeping to be done and this class solves that by doing it for you
+
+        The main idea is that we store a mask (labeeld_mask) of indices that have been labeled and then expose UD , LD
+        and the labels by using fancy indexing with that mask. The manager exposes a an add_labels method which lets the
+        user add labels indexed with respect to UD and it will adjust the indices so that they match OD.
+
+        :param features: An array of the features that will be used for AL.
+        :param labels: Any prexesiting labels. Each label is a tuple(idx,label)
+        :param source: A list of the original data
+        """
+        self.features = features
+
+        self._labels =np.empty(shape=self.features.shape[0],dtype=labels_dtype)
+        self.labeled_mask = np.zeros(self.features.shape[0], dtype=bool)
+        self.sources = np.array(sources if sources else [])
+
+    @property
+    def labels(self):
+        '''
+
+        Returns the labels indexed with respect to LD
+
+        '''
+        return self._labels[self.labeled_mask]
```

Comment:
This can change the data type of labels, if `labels_dtype` is not explicitly given upon initialization. How about adding a new attribute to the class where we store the label_dtype? This can be `None` by default and set during the `add_labels` method, so this method can return `self._labels[self.labeled_mask].astype(self.label_dtype)`. What do you think?